### PR TITLE
Ee 9349 split on full stops

### PR DESCRIPTION
**Pull request by owainrobertsbjss**
Updated the SpecialCharacter Name Matching Candidate Generator to strip out full stops as well as hyphens and apostrophes.

### CODE REVIEW

File: src/test/java/uk/gov/digital/ho/pttg/application/namematching/candidates/SpecialCharactersTest.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SpecialCharactersTest {
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/pttg/application/namematching/candidates/SpecialCharactersTest.java]: Should we have one test with multiple types of special characters to prove they work in conjunction?

File: src/test/java/uk/gov/digital/ho/pttg/application/namematching/candidates/SpecialCharactersTest.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/pttg/application/namematching/candidates/SpecialCharactersTest.java]: Also what do we do about special characters in aliases?

File: src/test/java/uk/gov/digital/ho/pttg/application/namematching/candidates/SpecialCharactersTest.java (hunk 1 above)
Comment (owainrobertsbjss) [reply src/test/java/uk/gov/digital/ho/pttg/application/namematching/candidates/SpecialCharactersTest.java]: Yes - I'll add that

File: src/test/java/uk/gov/digital/ho/pttg/application/namematching/candidates/SpecialCharactersTest.java (hunk 2 above)
Comment (owainrobertsbjss) [reply src/test/java/uk/gov/digital/ho/pttg/application/namematching/candidates/SpecialCharactersTest.java]: Currently nothing but it feels like we should handle in a similar manner to non-alias names. I have asked @ashanperera89 to confirm this requirement.

Comment (owainrobertsbjss) [reply src/test/java/uk/gov/digital/ho/pttg/application/namematching/candidates/SpecialCharactersTest.java]: Confirmed by @ashanperera89 that we should remove when the name has aliases, we should include them in the SpecialCharacter treatment. As is the case when we don't have special characters, the ordering of candidates should be different depending on whether or not aliases exist, and should follow the ordering rules already implemented.

**Pull request merged**